impl/src/planning/subplanners.rs

```diff
@@ -323,48 +350,30 @@ impl<'planner> CrateSubplanner<'planner> {
     let mut targets = self.produce_targets(&package_root)?;
     let build_script_target_opt = self.take_build_script_target(&mut targets);
 
-    let mut lib_target_name = None;
-    {
-      for target in &targets {
-        if target.kind == "lib" || target.kind == "proc-macro" {
-          lib_target_name = Some(target.name.clone());
-          break;
-        }
-      }
-    }
+    let lib_target_name = targets
+      .iter()
+      .find(|target| target.kind == "lib" || target.kind == "proc-macro")
+      .map(|target| target.name.clone());
 
+    let mut deps = self.produce_deps()?;
+    // Take the default deps that are not bound to platform targets
+    let default_deps = deps.remove(&None).unwrap_or_default();
     // Build a list of dependencies while addression a potential whitelist of target triples
-    let mut filtered_deps = Vec::new();
-    for dep_set in targeted_deps.iter() {
-      let mut target_triples = get_matching_bazel_triples(&dep_set.target)?;
-      filter_bazel_triples(
-        &mut target_triples,
-        self
-          .settings
-          .targets
-          .as_ref()
-          .unwrap_or(&Vec::<String>::new()),
-      );
-
-      if target_triples.len() == 0 {
-        continue;
-      }
-
-      filtered_deps.push(CrateTargetedDepContext {
-        target: dep_set.target.clone(),
-        deps: CrateDependencyContext {
-          dependencies: dep_set.dependencies.normal_deps.clone(),
-          proc_macro_dependencies: dep_set.dependencies.proc_macro_deps.clone(),
-          build_dependencies: dep_set.dependencies.build_deps.clone(),
-          build_proc_macro_dependencies: dep_set.dependencies.build_proc_macro_deps.clone(),
-          dev_dependencies: dep_set.dependencies.dev_deps.clone(),
-          aliased_dependencies: dep_set.dependencies.aliased_deps.clone(),
-        },
-        conditions: generate_bazel_conditions(&target_triples)?,
-      });
-    }
-
-    filtered_deps.sort();
+    let mut targeted_deps = deps
+      .into_iter()
+      .map(|(target, deps)| {
+        let target = target.unwrap().to_string();
+        let platform_targets = get_matching_bazel_triples(&target, &self.settings.targets)?
```

Comment:
I noticed in working with this, the targetting functions are a bit scattered and not really that reused.

As such I removed a bunch of the additional functions, smushing things down towards a single function, and in avoiding allocation-o-rama (which we do a bit frequently in cargo-raze) made it return an iterator.

The part for the generation of bazel conditions I moved into the template itself, as it seemed like a lot of code was written just to essentially prepend a string onto the platform targets.

Reply:
Yeah, I kinda ignore it in cargo-raze because the tool runs for me a few times a week, but when calling three consecutive functions seems to produce vector and string of each other it gets a bit much ;)